tracer/src/Datadog.Trace/AppSec/Waf/Initialization/WafConfigurator.cs

```diff
@@ -18,30 +20,55 @@ internal static class WafConfigurator
     {
         private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(WafConfigurator));
 
-        internal static IntPtr? Configure(string rulesFile, WafNative wafNative, Encoder encoder)
+        internal static InitializationResult Configure(string rulesFile, WafNative wafNative, Encoder encoder)
         {
             var argCache = new List<Obj>();
             var configObj = GetConfigObj(rulesFile, argCache, encoder);
             if (configObj == null)
             {
-                return null;
+                return InitializationResult.FromUnusableRuleFile();
```

Comment:
I think we could also log an error message here since the other log statements below won't be hit

Reply:
it's actually done inside of `GetConfigObj` , in `GetRulesStream` , and in the catch, there are no code paths inside of it, where null would come out of it without a log

Reply:
Makes sense, thanks for the clarification!